src/insert-extension.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getMode} from './mode';
+
+export let ampExtensionScriptInsertedOrPresent = [];
+
+/**
+ * Reset the ampExtensionScriptInsertedOrPresent value for each test.
+ * @visibleForTesting
+ */
+export function resetExtensionScriptInsertedOrPresentForTesting() {
+  ampExtensionScriptInsertedOrPresent = [];
+}
+
+/**
+ * Check script info in HTML head and make update if necessary
+ * @param {!Window} win
+ * @param {!Element} element
+ * @param {string} extension
+ */
+export function insertAmpExtensionScript/*OK*/(win, element, extension) {
+  if (extension == 'amp-embed') {
+    extension = 'amp-ad';
+  }
+  ampExtensionScriptInsertedOrPresent[extension] = false;
```

Comment:
Changed to an object. 
I previously thought `undefined` won't work for `if(ampExtensionScriptInsertedOrPresent)`. Already deleted the line.